### ISSUE TITLE
Unable to permanently remove groups or users from a folder.

### ISSUE THREAD

**Issue by JapstersCavern** [telegramdesktop/tdesktop#26894]
### Steps to reproduce

1. Right-click any Group or User, in lists / search results within the app.

2. Choose "Add To Folder" <folder name>.

3. Possibly, exit the app, if steps 4 onwards doe not reproduce the error (I don't know, because my Telegram now NEVER permanently  removes people, using the steps below.)

4. Right-click a Group or User, in lists / search results within the app, who shows as being in one of your folders (ticked).

5. Click on that user's ticked folder (the user will immediately disappear from that folder).

6. Right-click this SAME Group or User, in lists / search results within the app, who will now (correctly) NO LONGER show as being in the chosen folder.

7. *FULLY* EXIT Telegram, including via taskbar icon, and re-launch.

8. Search for the SAME Group or User, and right-click them again.

9. The user will STILL have the original folder ticked in "Add To Folders", and will STILL be in the folder(s) that they were just removed from.

### Expected behaviour

The user should be permanently removed from the chosen group(s), when Telegram Desktop is re-launched / computer power cycled..

### Actual behaviour

In between launches, the group/user's removal from a folder is not persistent, and reverts / changes are not kept (ADDITIONS to folder inclusion(s), for a group or user, however, ARE).

At present, this happens for ANY group / user, on my own installation, despite updates to the latest version of the Desktop App.

(It's fine for the current instance in memory, but reverts upon re-launching the app - changes are not persistent.)

I now have huge, no longer relevant folders with outdated contacts and groups in them...  :)

I must add, that this has now been happening for a LONG time, so the bug has been around for a long time, possibly around a year or more, but I just lived with it - felt I should log a bug, now I know that it can be done via GitHub...  :)

### Operating system

Windows 10 Home, 10.0.18362 (Build 18362)

### Version of Telegram Desktop

4.10.2 x64

### Installation source

Static binary from official website

**Comment by JapstersCavern** [comment]
Nope. Interesting approach. It's NOT fixed, just checked again and confirmed it, in the latest release of Telegram.

Removed a load of users from a folder.  Closed TG, opened it again, and they're all back.

Please don't rush to close something because a simple check to verify the issue hasn't been done!

I've tried this on several PCs too, and yep, it still happens. Either it's account-based, or TG-based - should be fairly simple to replicate, guys....

**Comment by JapstersCavern** [comment]
@dimas1980200  There's always one eh?...  ...Enjoy having that GitHub account reported and deleted for sending p0rn via GH's notification system...

**Comment by blueblynd** [comment]
I'm also having this issue — I delete the group out of the folder it's in, completely restart Telegram, and the group is back in the folder it should now be deleted from. Has there been any solving it yet? I regularly update Telegram, so I assume not, but just to check.

**Comment by 23rd** [comment]
Still an issue?

**Comment by JapstersCavern** [comment]
Yep. Just tried it again. Still an issue.

I remove a chat from one of my folders, seems fine. FULLY exit TG, and it's back again - happens for ANY chat I try it with, and any folder, chat, etc...

Surely this cannot be a hard fix to (a) reproduce, and (b) fix?...

**Comment by 23rd** [comment]
@JapstersCavern A while ago I added logging of an error possible related to your problem. Can you take a look at the `log.txt`?

**Comment by JapstersCavern** [comment]
> [@JapstersCavern](https://github.com/JapstersCavern) A while ago I added logging of an error possible related to your problem. Can you take a look at the `log.txt`?

Will take a look, cheers - I think I recall someone already saying to look in the log file a long time ago, and I didn't find anything useful - but here goes, thanks!...   :)

**Comment by alexb5dh** [comment]
Same issue.
The only interesting data I see in the logs is `"API Warning: not loaded minimal channel applied."` whenever I remove group from a folder.

**Comment by C0br4** [comment]
Issue still persists for me in 12/2025. On every restart, all groups that I moved to a different group (added to a new group and removed from the old one) appear in the old group as well after Telegram restart.

Operating system
Windows 11 Pro 23H2 (Build 22631.6199)

Version of Telegram Desktop
6.3.4 x64 Desktop

**Comment by JapstersCavern** [comment]
Hey guys, here's a groundbreaking idea! - why not...

...simply have this issue *FIXED*, instead of hoping it will just timeout until we miss the deadline to keep it open? - gonna be here watching this one on principle, until it's actually fixed (2-3 YEARS and counting, I believe now?).

It's incredibly annoying to keep removing groups, then have updates showing in folders, only to find it's yet another group I've ALREADY removed dozens of times already?

It's not instilling confidence in the QA of an otherwise excellent tool, team.

**Comment by Alatarius** [comment]
Still not fixed. Issue is still unresolved. Unable to delete any contacts that I have added regardless if in a group folder or not. This _should_ be an easy fix. But apparently they don't care to fix it since the issue has been around for 3 years now.

**Comment by reallyuniquename** [comment]
How many users did you exclude from a folder? 

Excluding users counts towards folder filtering limit. After you hit that it silently fails without any notification. This is why an excluded user pops up again in a folder after you restart Telegram Desktop.

**Comment by JapstersCavern** [comment]
> How many users did you exclude from a folder?
> 
> Excluding users counts towards folder filtering limit. After you hit that it silently fails without any notification. This is why an excluded user pops up again in a folder after you restart Telegram Desktop.

This may be, but would only be part of the problem / cause, in some cases, I believe - for example, look at my original post - even a new user can reproduce those steps, and hit the same issue - it's ridiculous that it's still happening, and being completely ignored, for almost THREE YEARS (and I've been paying a subscription for a LONG time - not really acceptable given we're helping fund the product).

...and to add, even creating a NEW folder, and adding a group or user to it, then removing them, simply does NOT work - they will ALWAYS appear in that folder again, after exiting and restarting TG...
